app/services/wiki-urls.js

```diff
@@ -159,6 +159,16 @@ export default Service.extend({
 		};
 	},
 
+	/**
+	 * Extracts the page title from a URL by stripping the host and article path.
+	 *
+	 * @param  {string} url - URL from which to extract the title
+	 * @return {string}
+	 */
+	getEncodedTitleFromURL(url) {
+		return url ? url.replace(new RegExp(`^(https?://[^/]+)?${this.langPathRegexp}?(/wiki)?/`), '') : '';
```

Comment:
`this.langPathRegexp` can be an empty string so this question mark will change the matcher from greedy to lazy (https://stackoverflow.com/a/21289386/1050577) on wikis without the lang path.